Clear information about overwriting configuration keys (#1533)
jenkinsci/configuration-as-code-plugin: README.md

:x: `/jenkins/.configs/casc_configs/..dir2/config.yaml`  
</details>

> Configuration file sets traversed in folders cannot overwrite configuration values, this will create a conflict and raise a `ConfiguratorException`. Thus, the order of traversal does not matter to the final outcome.

Author: ```suggestion
All configuration files that are discovered MUST be supplementary. They cannot overwrite each other's configuration values. This creates a conflict and raises a `ConfiguratorException`. Thus, the order of traversal does not matter to the final outcome.
```

Author: Feel free to use this alternative phrasing if it seems more clear.

Reviewer: +1 for alternate version. Thanks!
